helix-core/src/main/java/org/apache/helix/manager/zk/ZNRecordJacksonSerializer.java

```diff
@@ -0,0 +1,71 @@
+package org.apache.helix.manager.zk;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.I0Itec.zkclient.exception.ZkMarshallingError;
+import org.I0Itec.zkclient.serialize.ZkSerializer;
+import org.apache.helix.HelixException;
+import org.apache.helix.ZNRecord;
+import org.codehaus.jackson.map.ObjectMapper;
```

Comment:
How performance is the objectmapper, and AFAIK `codehaus` is a deprecated package

Reply:
Not great. I initially used the better-maintained half (fasterxml jackson) with MessagePack serialization, but hit a few issues in terms of backward compatibility and problems around deserializing. 

So I decided to just make it work first and optimize serialization later.

Reply:
Cool, thanks for the explanation.